### PR TITLE
ProcMount 1.30 update

### PR DESCRIPTION
**Pull request by haircommander**
KEP: https://github.com/kubernetes/enhancements/pull/4265

### CODE REVIEW

**Comment by Vyom-Yadav** [comment]
Hey @haircommander 👋 please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday March 12th 2024 18:00 PST. Thank you!

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:477]: The grammar doesn't look right to me.
I suggest:
```suggestion
## Managing access to the `/proc` filesystem {#proc-access}
```

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:481]:
```suggestion
For runtimes that follow the OCI runtime specification, containers default to running in a mode where
there are multiple paths that are both
```

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:487]: Avoid linking to the source code; instead, if we want to put a list in, pop it inside a new (small) page within https://kubernetes.io/docs/reference/node/

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:510]: (nit) Try
```suggestion
For some Pods, you might want to bypass that default masking of paths.
The most common context for wanting this is if you are trying to run containers within
a Kubernetes container (within a pod).
```

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:482]: I wouldn't put that link here, although I **would** put it inside the further reading section.

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:499]: We might want to call out this one especially, as it's outside `/proc`.

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:525]:
```suggestion
{{ < note > }}
In Kubernetes {{< skew currentVersion >}}, this field requires the `spec.hostUsers` value in the pod
spec to be `false`. In other words: a container that wishes to have an Unmasked
`/proc` or unmasked `/sys` must also be in a
[user namespace](/docs/concepts/workloads/pods/user-namespaces/).

Kubernetes v1.12 to v1.29 did not enforce that requirement.
{{< /note >}}
```

If I want to map `/sys/firmware` into a (privileged) Pod, and I'm using Kubernetes v1.30, how do I do that?

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:573]:
```suggestion
* Read about [user namespaces](/docs/concepts/workloads/pods/user-namespaces/) for Linux pods.
```

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:518]: We should give the reader more context about what to change; this looks pretty terse.

It's OK for alpha but would need to change before graduation to beta.

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:479]: Even better:
```suggestion
{{< feature-state feature_gate_name="ProcMountType" >}}
```

**Review by sftim — COMMENTED** [review]
Thanks; it'll need a few changes before we can merge this.

**Comment by sftim** [comment]
Please also fix the merge conflict (rebase against _dev-1.30_)

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (Vyom-Yadav) [reply content/en/docs/tasks/configure-pod-container/security-context.md:479]: @haircommander I don't want to add unnecessary noise, but I thought [KEP-4265](https://github.com/kubernetes/enhancements/issues/4265) is targeting `beta`. Am I missing something over here? Are these the missing docs for `alpha`? Where are the docs for `beta`?

**Comment by Vyom-Yadav** [comment]
Hello @mrunalp :wave: please provide a technical review for this PR by March 26th 2024 18:00 PT to get this into the release. Thank you!

**Comment by Vyom-Yadav** [comment]
Hey @haircommander @mrunalp 👋🏼 

I'm reaching out from the Docs team. Just checking in as we approach Docs Freeze on `Tuesday March 26th 18:00 PDT`. This documentation appears to still be under review. To meet the Docs Freeze, this PR must have a technical review as well as `lgtm` and `approve` labels applied, without any unaddressed comments or concerns from SIG Docs. The status of this enhancement is marked as `at risk for docs freeze`. Thank you!

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (haircommander) [reply content/en/docs/tasks/configure-pod-container/security-context.md:537]: if a pod is privileged then no paths are masked

**Review by mrunalp — APPROVED** [review]
/approve
Technically LGTM from sig-node.

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (haircommander) [reply content/en/docs/tasks/configure-pod-container/security-context.md]: how do you feel about me dropping the link to the source code but not adding a reference? having it in 3 places seems more likely to skew eventually

**Comment by drewhagen** [comment]
Just a heads up that Doc Freeze is approaching fast tomorrow March 26th at 18:00 PDT and starting this cycle, we need to file [an Exception](https://github.com/kubernetes/sig-release/blob/master/releases/EXCEPTIONS.md) after this time.
This one looks pretty close and we'd like to help you get it over the finish line!

@kubernetes/sig-docs-en-owners Does this one look good for an `approve` from y'all? Thanks!

**Review by dipesh-rawat — COMMENTED** [review]
@haircommander The PR is currently lacking the doc change to transition the feature gate from Alpha to Beta. The necessary change should be made in the feature gate description file - `proc-mount-type.md` ([here](https://github.com/kubernetes/website/blob/dev-1.30/content/en/docs/reference/command-line-tools-reference/feature-gates/proc-mount-type.md)).

Please refer to the documentation [here](https://kubernetes.io/docs/contribute/new-content/new-features/#ready-for-review-feature-gates) for further information.

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (Vyom-Yadav) [reply content/en/docs/tasks/configure-pod-container/security-context.md]: Oh, I see. The alpha feature was merged quite some time ago, but there weren't any docs for the feature. This PR just simply adds docs for that feature gate.

**Comment by Vyom-Yadav** [comment]
> The PR is currently lacking the doc change to transition the feature gate from Alpha to Beta.

@dipesh-rawat I don't think PR is targeting beta. [KEP-4265](https://github.com/kubernetes/enhancements/issues/4265) was updated to `stage/alpha`. So the feature gate will not be bumped. Rest @haircommander can confirm.

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:481]: Is this right?

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:487]:
```suggestion
```

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/security-context.md:537]: OK, how about
```suggestion
{{ < note > }}
Setting `procMount` to Unmasked requires the `spec.hostUsers` value in the pod
spec to be `false`. In other words: a container that wishes to have an Unmasked
`/proc` or unmasked `/sys` must also be in a
[user namespace](/docs/concepts/workloads/pods/user-namespaces/).

Kubernetes v1.12 to v1.29 did not enforce that requirement.
{{< /note >}}
```
?

**Comment by sftim** [comment]
Also see https://github.com/kubernetes/website/pull/45152#discussion_r1539276457

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/security-context.md]: That's fine.

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/security-context.md:509]: What I mean is: people might think that  `procMount` only covers `/proc`.

Comment (haircommander) [reply content/en/docs/tasks/configure-pod-container/security-context.md:509]: I left a note below does that suffice?

**Comment by haircommander** [comment]
> > The PR is currently lacking the doc change to transition the feature gate from Alpha to Beta.
> 
> @dipesh-rawat I don't think PR is targeting beta. [KEP-4265](https://github.com/kubernetes/enhancements/issues/4265) was updated to `stage/alpha`. So the feature gate will not be bumped. Rest @haircommander can confirm.

correct! we're not bumping this to beta in this release

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (haircommander) [reply content/en/docs/tasks/configure-pod-container/security-context.md:481]: ah this was pulled in from a merge conflict...

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (drewhagen) [reply content/en/docs/tasks/configure-pod-container/security-context.md]: Looking at the diff, it looks like this change was made; this can be resolved.

**Comment by drewhagen** [comment]
Looks like feedback was addressed. @kubernetes/sig-docs-en-owners Is this ready to approve?

**Comment by reylejano** [comment]
Looks like feed back has been addressed and there is a technical lgtm from SIG Node
/approve

File: content/en/docs/tasks/configure-pod-container/security-context.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/security-context.md:578]: Ideally we'd have fixed this broken link before a merge. I'll file an issue.

Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/security-context.md:578]: Actually, I'll just send in a PR.

**Pull request merged**
